### PR TITLE
install wpasupplicant when a wifi nic is seen

### PR DESCRIPTION
**Pull request by mwhudson**
This should (together with seed changes) should be enough to support wifi in subiquity. Haven't tested it yet though!

### CODE REVIEW

**Comment by mwhudson** [comment]
So with https://github.com/canonical/subiquity/pull/980 and https://github.com/canonical/probert/pull/106 (and https://code.launchpad.net/~mwhudson/casper/+git/casper/+merge/403846 and https://code.launchpad.net/~mwhudson/ubuntu-seeds/+git/ubuntu/+merge/403848) this works, at least as far as configuring networking in the installer environment! (I don't have a machine to hand I can actually install to to test that end of things, although that is a fair bit simpler)

**Comment by xnox** [comment]
```
Testing WiFi in VM
* Boot to Try Ubuntu
* modprobe virt_wifi module
* Use `ip link add link en0 name wlan0 type virt_wifi` to wrap wifi in ethernet
```

File: subiquity/models/network.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 class NetworkModel(NetworkModel):
```
Comment (dbungert) [inline subiquity/models/network.py:23]: Wait, that works?  That looks super strange.
I know this isn't part of the PR but what's your thoughts on a minor rename?

File: subiquity/server/controllers/install.py [hunk 2]
```diff
@@ -268,6 +268,8 @@ async def postinstall(self, *, context):
         packages = []
         if self.model.ssh.install_server:
             packages = ['openssh-server']
```
Comment (dbungert) [inline subiquity/server/controllers/install.py:270]: Again, not part of the PR, but maybe we should packages.append() instead of assigning directly.

File: subiquitycore/ui/views/network.py [hunk 3]
```diff
@@ -409,9 +431,31 @@ def _create_bond(self, sender=None):
         stretchy.attach_context(self.controller.context.child("add_bond"))
         self.show_stretchy_overlay(stretchy)
 
+    def update_for_wlan_support_install_state(self, state):
+        if state == "NOT_NEEDED":
+            return
+        if self.error_showing:
+            b = 2
```
Comment (dbungert) [inline subiquitycore/ui/views/network.py:438]: What about instead of these b/i variables, we define some start and end points?  I think that would scan better.  What do you think?  Also I had no idea I could define a slice ahead of time, really cool.

**Review by dbungert — CHANGES_REQUESTED** [review]
I do have some suggestions but they are mostly on code not even in this PR, so do what you will with that.

File: subiquity/models/network.py (hunk 1 above)
Comment (mwhudson) [reply subiquity/models/network.py:23]: Yeah it works fine (welcome to Python!). It might be a bit strange but can we leave it for now? :)

File: subiquity/server/controllers/install.py (hunk 2 above)
Comment (mwhudson) [reply subiquity/server/controllers/install.py:270]: yeah i thought that when writing it. will change.

File: subiquitycore/ui/views/network.py (hunk 3 above)
Comment (mwhudson) [reply subiquitycore/ui/views/network.py:438]: yeah that would be clearer, I'll change this too.

**Review by dbungert — APPROVED** [review]
Thanks!
